src/cupbearer/tasks/backdoor_detection.py

```diff
@@ -1,34 +1,39 @@
-from copy import deepcopy
 from dataclasses import dataclass
 from pathlib import Path
 
-from cupbearer.data import Backdoor
+from cupbearer.data import DatasetConfig
 from cupbearer.data.backdoor_data import BackdoorData
-from cupbearer.models import StoredModel
+from cupbearer.models import ModelConfig, StoredModel
 from cupbearer.utils.scripts import load_config
 
-from ._config import DebugTaskConfig, TaskConfig
+from ._config import DebugTaskConfig, FuzzedTask
 
 
 @dataclass(kw_only=True)
-class BackdoorDetection(TaskConfig):
+class BackdoorDetection(FuzzedTask):
     path: Path
-    backdoor: Backdoor
     no_load: bool = False
 
-    def _init_train_data(self):
-        data_cfg = load_config(self.path, "train_data", BackdoorData)
-        # Remove the backdoor
-        self._train_data = data_cfg.original
+    def __post_init__(self):
+        backdoor_data = load_config(self.path, "train_data", BackdoorData)
+        self._original = backdoor_data.original
+        self._backdoor = backdoor_data.backdoor
```

Comment:
The name change from `backdoor` to `_backdoor` breaks the `test_wanet` test. The test should be updated.